src/bp/core/services/action/action-service.ts

```diff
@@ -152,35 +162,66 @@ export class ScopedActionService {
     return !!actions.find(x => x.name === actionName)
   }
 
-  private _prepareRequire(fullPath: string) {
-    const actionLocation = path.dirname(fullPath)
+  async getActionDetails(actionName: string) {
+    const action = await this.findAction(actionName)
+    const code = await this.getActionScript(action)
+
+    const botFolder = action.location === 'global' ? 'global' : 'bots/' + this.botId
+    const dirPath = path.resolve(path.join(process.PROJECT_LOCATION, `/data/${botFolder}/actions/${actionName}.js`))
+    const lookups = getBaseLookupPaths(dirPath)
+
+    return { code, dirPath, lookups, action }
+  }
+
+  // This method tries to load require() files from the FS and fallback on BPFS
+  async checkActionRequires(actionName: string): Promise<boolean> {
+    if (this._validScripts[actionName]) {
+      return true
+    }
+
+    const { code, dirPath: parentScript, lookups } = await this.getActionDetails(actionName)
 
-    let parts = path.relative(process.PROJECT_LOCATION, actionLocation).split(path.sep)
-    parts = parts.slice(parts.indexOf('actions') + 1) // We only keep the parts after /actions/...
+    const isRequireValid = prepareRequireTester(parentScript, lookups)
+    const files = extractRequiredFiles(code)
 
-    const lookups: string[] = [actionLocation]
+    for (const file of files) {
+      if (isRequireValid(file)) {
```

Comment:
what's that ? if the require is valid, you skip the actual require, maybe it's just a naming issue

Reply:
If the file was required successfully, then it doesn't need to try to load it from the bpfs